src/topology/continuous_function/continuous_section.lean

```diff
@@ -0,0 +1,51 @@
+/-
+Copyright © 2021 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nicolò Cavalleri
+-/
+
+import topology.continuous_function.basic
+import data.right_inv
+
+/-!
+# Continuous sections
+In this file we define bundled continuous sections of a map.
+-/
+
+set_option old_structure_cmd true
+
+variables {α: Type*} {β: Type*} {f : α → β}
+variables [topological_space α] [topological_space β]
+
+/-- A continuous section of a map `f` is a continuous map that is a right inverse of `f`.
+Note that the `nolint has_inhabited_instance` here has a true matematical meaning: the universal
+cover of a circle is a map that does not admit continuous sections. -/
+@[nolint has_inhabited_instance]
+structure continuous_section (f : α → β) extends right_inv f, C(β, α)
+
+instance : has_coe_to_fun (continuous_section f) := ⟨λ f, β → α, continuous_section.to_fun⟩
+
+instance : has_coe (continuous_section f) (right_inv f) :=
+⟨continuous_section.to_right_inv⟩
```

Comment:
You should add a simp lemma that rewrites `continuous_section.to_right_inv = coe` or vice versa